File: backend/src/main/java/com/magesty/backend/models/Education.java
package com.magesty.backend.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.magesty.backend.models.dto.AdminDto;
import com.magesty.backend.models.dto.EducationDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.time.LocalDate;
import java.util.Objects;
import java.util.UUID;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class Education implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String etablissement;
    private String section;
    private String degree;
    private String average;
    private LocalDate start;
    private LocalDate expected_end;
    private String image;
    private String current;

    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn( name = "eduId")
    private Admin admin;

    public static Education from(EducationDto educationDto){
        Education education = new Education();

        if(Objects.isNull(educationDto)){
            return null;
        }else {
            education.setId(educationDto.getId());
            education.setDegree(educationDto.getDegree());
            education.setAverage(educationDto.getAverage());
            education.setSection(educationDto.getSection());
            education.setEtablissement(educationDto.getEtablissement());
            education.setCurrent(educationDto.getCurrent());
            education.setStart(educationDto.getStart());
            education.setExpected_end(educationDto.getExpected_end());
            education.setImage(educationDto.getImage());

            return education;
        }
    }
}

File: frontend-ng/src/app/models/Education/education.ts
export class Education {
  private _id: number;
  private _etablissement: string;
  private _degree: string;
  private _section: string;
  private _average: string;
  private _start: Date;
  private _expected_end: Date;
  private _current: string;
  private _image: string;

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get etablissement(): string {
    return this._etablissement;
  }

  get section(): string {
    return this._section;
  }

  set section(value: string) {
    this._section = value;
  }

  set etablissement(value: string) {
    this._etablissement = value;
  }

  get degree(): string {
    return this._degree;
  }

  set degree(value: string) {
    this._degree = value;
  }

  get average(): string {
    return this._average;
  }

  set average(value: string) {
    this._average = value;
  }

  get start(): Date {
    return this._start;
  }

  set start(value: Date) {
    this._start = value;
  }

  get expected_end(): Date {
    return this._expected_end;
  }

  set expected_end(value: Date) {
    this._expected_end = value;
  }

  get current(): string {
    return this._current;
  }

  set current(value: string) {
    this._current = value;
  }

  get image(): string {
    return this._image;
  }

  set image(value: string) {
    this._image = value;
  }
}
